plugins/monitoring/app/controllers/monitoring/alarm_definitions_controller.rb

```diff
@@ -296,17 +301,25 @@ def dimension_row
 
     private
     
-    def parse_expression(expression)
+    def parse_expression(expression_original)
 
       # at the moment period, statistical function and dimensions are optional
       #  - period is set with 60 seconds, if it is not existing
       #  - statistical function is set with avg, if it is not existing
       # metric, threshold and threshold value are required
       
-      # remove all white spaces
+      expression = expression_original.clone
+      # check for compounded expression
+      if expression =~ /( or | and )/i
```

Comment:
I would suggest to use `\s` instead of a space, just in case someone tries to be clever and uses tabs or (more likely) newlines to format their alert expression.